cshared.go

```diff
@@ -130,30 +131,16 @@ func FLBPluginInputCallback(data *unsafe.Pointer, csize *C.size_t) int {
 		return input.FLB_RETRY
 	}
 
-	var err error
-	once.Do(func() {
-		runCtx, runCancel = context.WithCancel(context.Background())
-		theChannel = make(chan Message)
-		go func() {
-			err = theInput.Collect(runCtx, theChannel)
-		}()
-	})
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "run: %s\n", err)
-		return input.FLB_ERROR
-	}
-
-	select {
-	case msg, ok := <-theChannel:
-		if !ok {
-			return input.FLB_OK
-		}
+	var mu sync.Mutex
+	send := func(msg Message) {
+		mu.Lock()
+		defer mu.Unlock()
 
 		t := input.FLBTime{Time: msg.Time}
 		b, err := input.NewEncoder().Encode([]any{t, msg.Record})
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "encode: %s\n", err)
-			return input.FLB_ERROR
+			return
 		}
 
 		cdata := C.CBytes(b)
```

Comment:
Each time send is executed another buffer will be leaked. Only a single buffer will be received by the calyptia side in each polling interval (1000 nanoseconds). Only the buffer that actually gets received will be freed.